settings/src/components/backup-codes.js

```diff
@@ -41,25 +40,20 @@ export default function BackupCodes() {
 		);
 	}
 
-	if (
-		! backupCodesEnabled ||
-		backupCodesRemaining === 0 ||
-		regenerating ||
-		! backupCodesVerified
```

Comment:
These are the conditions which led to the codes being generated on load

Reply:
We currently auto navigate to the backup code component when a user sets up a one time password. We'll also do that in the onboarding [flow](https://github.com/WordPress/wporg-two-factor/pull/297) and add that support for [Webauthn](https://github.com/WordPress/wporg-two-factor/issues/253). Now that this doesn't auto trigger, do you think we should export the "Setup" component and handle that in those flows? Alternatively we could pass a prop into this component. Thoughts?

Reply:
Ah I wasn't aware that they need to be auto generated in those cases. Yeah passing a prop seems reasonable.

Reply:
Sorry I'm confused about this. We can still auto navigate to backup codes, but my understanding is that we still should not auto generate. Is it a requirement that codes are generated if one of the 2FA methods has been activated? If it is then auto generation still makes sense.

Reply:
It's not a requirement per se, but we want to limit the number of accounts that don't have recovery codes as much as possible seeing that regaining access to a locked out account can be troublesome for users.

Reply:
Yeah I guess it's a balance between acting on their account without their action vs ensuring they don't get locked out. I'll add the auto-generation so we can see what it looks like.

Reply:
Reading through #507, and in particular @dd32's [comment](https://github.com/WordPress/two-factor/issues/507#issuecomment-1425108260), I don't think auto generating is the best path here. If we're going to auto generate it feels like we should only display the codes, and require a second action to save. I've updated the button text in this PR to be more explicit about the action taken if they do choose to generate. What do you think?

Reply:
I think that's fair. I just tested this. It does help to establish context. 

We can monitor whether users bail on this view and adjust as necessary.